src/main/java/org/projectnessie/tool/CatalogMigrateUtil.java

```diff
@@ -0,0 +1,174 @@
+    return migrateTables(tableIdentifiers, sourceCatalog, targetCatalog, maxThreadPoolSize, true);
+  }
+
+  /**
+   * Register tables from one catalog(source catalog) to another catalog(target catalog). User has
+   * to take care of deleting the tables from source catalog after registration.
+   *
+   * <p>Supports bulk registration with a multi-thread execution.
+   *
+   * <p>Users must make sure that no in-progress commits on the tables of source catalog during
+   * registration.
+   *
+   * @param tableIdentifiers a list of {@link TableIdentifier} for the tables required to be
+   *     registered. If not specified, all the tables would be registered
+   * @param sourceCatalog Source {@link Catalog} from which the tables are chosen
+   * @param targetCatalog Target {@link Catalog} to which the tables need to be registered
+   * @param maxThreadPoolSize Size of the thread pool used for registering tables (If set to 0, no
+   *     thread pool is used)
+   * @return Collection of table identifiers for successfully registered tables
+   */
+  public static Collection<TableIdentifier> registerTables(
+      List<TableIdentifier> tableIdentifiers,
+      Catalog sourceCatalog,
+      Catalog targetCatalog,
+      int maxThreadPoolSize) {
+    return migrateTables(tableIdentifiers, sourceCatalog, targetCatalog, maxThreadPoolSize, false);
+  }
+
+  private static Collection<TableIdentifier> migrateTables(
+      List<TableIdentifier> tableIdentifiers,
+      Catalog sourceCatalog,
+      Catalog targetCatalog,
+      int maxThreadPoolSize,
+      boolean deleteEntriesFromSourceCatalog) {
+    validate(sourceCatalog, targetCatalog, maxThreadPoolSize);
+
+    List<TableIdentifier> identifiers;
+    if (tableIdentifiers == null || tableIdentifiers.isEmpty()) {
+      // fetch all the table identifiers from all the namespaces.
+      List<Namespace> namespaces =
+          (sourceCatalog instanceof SupportsNamespaces)
+              ? ((SupportsNamespaces) sourceCatalog).listNamespaces()
+              : ImmutableList.of(Namespace.empty());
+      identifiers =
+          namespaces.stream()
+              .flatMap(namespace -> sourceCatalog.listTables(namespace).stream())
+              .collect(Collectors.toList());
+    } else {
+      identifiers = tableIdentifiers;
+    }
+
+    ExecutorService executorService = null;
+    if (maxThreadPoolSize > 0) {
+      executorService = ThreadPools.newWorkerPool("migrate-tables", maxThreadPoolSize);
+    }
+
+    try {
+      Collection<TableIdentifier> migratedTableIdentifiers = new ConcurrentLinkedQueue<>();
+      Tasks.foreach(identifiers.stream().filter(Objects::nonNull))
+          .retry(3)
+          .stopRetryOn(NoSuchTableException.class, NoSuchNamespaceException.class)
+          .suppressFailureWhenFinished()
+          .executeWith(executorService)
+          .onFailure(
+              (tableIdentifier, exc) ->
+                  LOG.warn("Unable to migrate table {}", tableIdentifier, exc))
```

Comment:
I don't want this to be StdOut or pass printWriter here as these API can be used independently of CLI too. Hence, the log.